catalyst_voices/lib/widgets/snackbar/voices_snackbar_type.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:catalyst_voices_brands/catalyst_voices_brands.dart';
+import 'package:catalyst_voices_localization/catalyst_voices_localization.dart';
+import 'package:flutter/material.dart';
+
+enum VoicesSnackBarType { info, success, warning, error }
+
+class _SnackBarData {
+  final IconData icon;
+  final Color? Function(VoicesColorScheme) iconColor;
+  final Color? Function(VoicesColorScheme) titleColor;
+  final Color? Function(VoicesColorScheme) backgroundColor;
+  final String Function(VoicesLocalizations) message;
+  final String Function(VoicesLocalizations) title;
```

Comment:
I imagine we'll define more functions like this one. Especially when localising errors i think we may want to define it somewhere. Maybe in `catalyst_voices_localization`?


`catalyst_voices_localization/lib/src/typedefs.dart`

```dart
import 'package:flutter/widgets.dart';

/// Example (context) => context.l10n.snackbarInfoLabelText
typedef LocalizationResolver = String Function(BuildContext context);
```

Same with colors

Reply:
That's a good idea. However, `LocalizationResolver ` is too long. 
How about `L10n`? Do you have any ideas for colors?

Reply:
`L10nResolver` sounds good to me 👍 

for colors i would say just `ColorResolver` because it may be coming from `VoicesColorScheme`, `ColorScheme` or even just hardcoded.

Reply:
Ideally, we should use only `VoicesColorScheme `.